src/backend/opencl/kernel/reduce_blocks_by_key_dim.cl

```diff
@@ -7,7 +7,7 @@
  * http://arrayfire.com/licenses/BSD-3-Clause
  ********************************************************/
 
-Tk work_group_scan_inclusive_add(__local Tk *wg_tmp, __local Tk *arr) {
+Tk wgScanInclusiveAdd(__local Tk *wg_tmp, __local Tk *arr) {
```

Comment:
If we can mimic the same functionality we should be able to just ifdef this out.

Reply:
Yeah, true - the thought crossed my mind. I think I saw different signatures, so postponed it for later wondering if we should push fix release soon :thinking:

Reply:
I believe we can't declare local memory inside a non-kernel function in OpenCL version we develop 1.2 or lower. I have to verify this, but I remember handling this few weeks back in another kernel.

If we can't do the above, I don't think we can mimic https://www.khronos.org/registry/OpenCL/sdk/2.0/docs/man/xhtml/work_group_reduce.html API, because it takes a single scalar much like sync_threads_or in CUDA.

Reply:
Yeah, that wouldn't be possible in 1.2. nevermind.

Reply:
Can you think of a way to implement the alternative 2.0 code path for this kernel?

Reply:
Let me look into it.